src/app/lib/utilities/preprocessor/rules/addIdsToItems.js

```diff
@@ -0,0 +1,31 @@
+import uuid from 'uuid';
+import compose from 'ramda/src/compose';
+import path from 'ramda/src/path';
+
+const getGroups = path(['content', 'groups']);
+
+const addIdToItem = ({ id, ...item }) => ({ ...item, id: id || uuid() });
+
+const addIdsToGroupItems = ({ items, ...group }) => ({
+  ...group,
+  items: items && items.map(addIdToItem),
+});
+
+const mapGroups = groups => groups.map(addIdsToGroupItems);
+
+const mergeContentWithAddedIdItems = itemsWithIds => jsonRaw => ({
+  ...jsonRaw,
+  content: {
```

Comment:
I think it would be safer to also spread the `jsonRaw.content` properties here so that no data is lost e.g.

```js
  ...jsonRaw,
  content: {
    ...path(['content'], jsonRaw),
    groups: itemsWithIds,
  },
```